client/src/test/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageOrderlyServiceTest.java

```diff
@@ -0,0 +1,362 @@
+        assertThat(resultNull.getConsumeResult()).isEqualTo(CMResult.CR_SUCCESS);
+    }
+
+
+    @Test
+    public void testConsumeMessageDirectly_Suspend() {
+        ConsumeMessageOrderlyService service = new ConsumeMessageOrderlyService(pushConsumer.getDefaultMQPushConsumerImpl(), new MessageListenerOrderly() {
+            @Override
+            public ConsumeOrderlyStatus consumeMessage(List<MessageExt> msgs, ConsumeOrderlyContext context) {
+                return ConsumeOrderlyStatus.SUSPEND_CURRENT_QUEUE_A_MOMENT;
+            }
+        });
+        MessageExt msg = new MessageExt();
+        ConsumeMessageDirectlyResult resultNull = service.consumeMessageDirectly(msg, broker);
+        assertThat(resultNull.getConsumeResult()).isEqualTo(CMResult.CR_LATER);
+    }
+
+
+    @Test
+    public void testConsumeMessageDirectly_Commit() {
+        ConsumeMessageOrderlyService service = new ConsumeMessageOrderlyService(pushConsumer.getDefaultMQPushConsumerImpl(), new MessageListenerOrderly() {
+            @Override
+            public ConsumeOrderlyStatus consumeMessage(List<MessageExt> msgs, ConsumeOrderlyContext context) {
+                return ConsumeOrderlyStatus.COMMIT;
+            }
+        });
+        MessageExt msg = new MessageExt();
+        ConsumeMessageDirectlyResult resultNull = service.consumeMessageDirectly(msg, broker);
+        assertThat(resultNull.getConsumeResult()).isEqualTo(CMResult.CR_COMMIT);
+    }
+
+
+    @Test
+    public void testConsumeMessageDirectly_Rollback() {
+        ConsumeMessageOrderlyService service = new ConsumeMessageOrderlyService(pushConsumer.getDefaultMQPushConsumerImpl(), new MessageListenerOrderly() {
+            @Override
+            public ConsumeOrderlyStatus consumeMessage(List<MessageExt> msgs, ConsumeOrderlyContext context) {
+                return ConsumeOrderlyStatus.ROLLBACK;
+            }
+        });
+        MessageExt msg = new MessageExt();
+        ConsumeMessageDirectlyResult resultNull = service.consumeMessageDirectly(msg, broker);
+        assertThat(resultNull.getConsumeResult()).isEqualTo(CMResult.CR_ROLLBACK);
+    }
+
+
+    @Test
+    public void testConsumeMessageDirectly_Exception() {
+        ConsumeMessageOrderlyService service = new ConsumeMessageOrderlyService(pushConsumer.getDefaultMQPushConsumerImpl(), new MessageListenerOrderly() {
+            @Override
+            public ConsumeOrderlyStatus consumeMessage(List<MessageExt> msgs, ConsumeOrderlyContext context) {
+                int r = 1 / 0;
+                return ConsumeOrderlyStatus.SUSPEND_CURRENT_QUEUE_A_MOMENT;
+            }
+        });
+        MessageExt msg = new MessageExt();
+        ConsumeMessageDirectlyResult resultNull = service.consumeMessageDirectly(msg, broker);
+        assertThat(resultNull.getConsumeResult()).isEqualTo(CMResult.CR_THROW_EXCEPTION);
+    }
+
+
+    @Test
+    public void testSubmitConsumeRequest() {
```

Comment:
We are verifying normal consumption in this test case，wo need   verify consume success!

Reply:
ok